pkg/ctl/create/iamserviceaccount.go

```diff
@@ -46,6 +46,7 @@ func createIAMServiceAccountCmdWithRunFunc(cmd *cmdutils.Cmd, runFunc func(cmd *
 		fs.StringVar(&serviceAccount.Namespace, "namespace", "default", "namespace where to create the iamserviceaccount")
 		fs.StringSliceVar(&serviceAccount.AttachPolicyARNs, "attach-policy-arn", []string{}, "ARN of the policy where to create the iamserviceaccount")
 		fs.StringVar(&serviceAccount.RoleName, "role-name", "", "Set a custom name for the created role")
+		serviceAccount.RoleOnly = fs.Bool("role-only", false, "disable service account creation, only the role will be created")
```

Comment:
Can we use the same format as the above lines for adding args? Independent of which it's better, I think we should keep the same style.

Reply:
`BoolVar` does not work, because it wants a bool pointer and not a pointer to a bool pointer, so I get a null pointer exception. That's because I use bool pointer in serviceAccount to model the absence of the RoleOnly option, maybe there is a better pattern.

Reply:
I think you can just pass in `serviceAccount.RoleOnly` and add `RoleOnly: api.Disabled(),` to https://github.com/weaveworks/eksctl/blob/7901d1eebe27beb2205922d56be1b72b4d49fa72/pkg/ctl/create/iamserviceaccount.go#L28